packages/frama-c/frama-c.26.0/opam

```diff
@@ -0,0 +1,171 @@
+- a runtime verification tool (E-ACSL plug-in);
+- several tools for code base exploration and dependency analysis
+  (plug-ins From, Impact, Metrics, Occurrence, Scope, etc.).
+These plug-ins communicate between each other via the Frama-C API
+and via ACSL (ANSI/ISO C Specification Language) properties.
+"""
+maintainer: "francois.bobot@cea.fr"
+authors: [
+  "Michele Alberti"
+  "Thibaud Antignac"
+  "Gergö Barany"
+  "Patrick Baudin"
+  "Thibaut Benjamin"
+  "Allan Blanchard"
+  "Lionel Blatter"
+  "François Bobot"
+  "Richard Bonichon"
+  "Quentin Bouillaguet"
+  "David Bühler"
+  "Zakaria Chihani"
+  "Loïc Correnson"
+  "Julien Crétin"
+  "Pascal Cuoq"
+  "Zaynah Dargaye"
+  "Basile Desloges"
+  "Jean-Christophe Filliâtre"
+  "Philippe Herrmann"
+  "Maxime Jacquemin"
+  "Florent Kirchner"
+  "Tristan Le Gall"
+  "Jean-Christophe Léchenet"
+  "Matthieu Lemerre"
+  "Dara Ly"
+  "David Maison"
+  "Claude Marché"
+  "André Maroneze"
+  "Thibault Martin"
+  "Fonenantsoa Maurica"
+  "Melody Méaulle"
+  "Benjamin Monate"
+  "Yannick Moy"
+  "Anne Pacalet"
+  "Valentin Perrelle"
+  "Guillaume Petiot"
+  "Dario Pinto"
+  "Virgile Prevosto"
+  "Armand Puccetti"
+  "Félix Ridoux"
+  "Virgile Robles"
+  "Muriel Roger"
+  "Julien Signoles"
+  "Nicolas Stouls"
+  "Kostyantyn Vorobyov"
+  "Boris Yakobowski"
+]
+homepage: "https://frama-c.com/"
+license: "LGPL-2.1-only"
+dev-repo: "git+https://git.frama-c.com/pub/frama-c.git"
+doc: "http://frama-c.com/download/user-manual-26.0-Iron.pdf"
+bug-reports: "https://git.frama-c.com/pub/frama-c/issues"
+tags: [
+  "deductive"
+  "program verification"
+  "formal specification"
+  "automated theorem prover"
+  "interactive theorem prover"
+  "C"
+  "plugins"
+  "abstract interpretation"
+  "slicing"
+  "weakest precondition"
+  "ACSL"
+  "dataflow analysis"
+  "runtime verification"
+]
+
+build: [
+  ["dune" "build" "-j%{jobs}%" "--release" "--promote-install-files=false" "@install"]
+  [make "-C" "doc" "download"] {with-doc}
+]
+
+install: [
+  [make "PREFIX=%{prefix}%" "MANDIR=%{mandir}%" "install"]
+  [make "PREFIX=%{prefix}%" "-C" "doc" "install"] {with-doc}
+]
+
+remove: [
+  [make "PREFIX=%{prefix}%" "-f" "ivette/Makefile.installation" "uninstall"]
+]
+
+run-test: [
+  ["dune" "exec" "--" "frama-c-ptests" "tests" "src/plugins/*/tests"
+  ] { arch != "ppc64" & arch != "x86_32" & arch != "arm32" & os != "macos" }
+  ["dune" "build" "-j%{jobs}%" "@ptests_config"
+  ] { arch != "ppc64" & arch != "x86_32" & arch != "arm32" & os != "macos" }
+]
+
+depends: [
+  "dune" { (>= "3.2.0" & os!="macos") | (>= "3.5.0" & os="macos") }
+  "dune-configurator"
+  "dune-private-libs"
+  "dune-site"
+
+  ( "alt-ergo-free" | "alt-ergo" )
+  "conf-graphviz" { post }
+  "conf-time" { with-test }
+  "ocaml" { >= "4.11.1" }
+  "ocamlfind" # needed beyond build stage, used by -load-module
+  "ocamlgraph" { >= "1.8.8" }
+  "why3" { >= "1.5.1" }
+  "yojson" { >= "1.6.0" & < "2.0.0" }
```

Comment:
Is that really the case? yojson 2 was released almost 6 months ago at this point so I’m just wondering if this is just not a remanent of an outdated state of frama-c
```suggestion
  "yojson" { >= "1.6.0" }
```

Reply:
In fact it is related to the tests. The pretty printer has been changed in Yojson 2.0.0, and it breaks some of our tests. Maybe we could indicate this restriction for the test only :thinking: . Does it make sense?

Reply:
ah indeed. In this case I'll change that for all the previous versions as well. It seems to have been a problem since frama-c 22.0. In `frama-c.25.0` the package added with `"yojson" {>= "1.6.0" & ( < "2.0.0" | ! with-test) }` but for some reason subsequent releases removed it. Do you know the reason behind it?

Reply:
```suggestion
  "yojson" {>= "1.6.0" & (< "2.0.0" | !with-test)}
```
I don't have write access to your branch. Could you apply this suggestion?

Reply:
> Do you know the reason behind it?

Hum, no, I don't know. Migrating Frama-C to Dune forced us to change a lot of things and maybe we just removed this constraint so that pinning dependencies for working on the dune development branch was faster. I'm not sure.